src/NuGet.Core/NuGet.Packaging/TopologicalSortUtility.cs

```diff
@@ -10,24 +10,21 @@ namespace NuGet.Packaging
 {
     public static class TopologicalSortUtility
     {
+        private static readonly PackageInfoComparer DefaultComparer = new PackageInfoComparer();
         /// <summary>
         /// Order dependencies by children first.
         /// </summary>
         public static IReadOnlyList<PackageDependencyInfo> SortPackagesByDependencyOrder(
             IEnumerable<PackageDependencyInfo> packages)
         {
-            var sorted = new List<PackageDependencyInfo>();
-            var toSort = packages.Distinct().ToList();
+            var toSort = packages.Distinct().Select(package => new PackageInfo(package)).ToArray();
```

Comment:
ToList seems better than ToArray here because it shouldn't require an additional copy allocation.